models.py

```diff
@@ -1,7 +1,137 @@
+    prod_name = db.Column(db.String(80))
+    new_qty = db.Column(db.Integer)
+    used_qty = db.Column(db.Integer)
+    open_boxed_qty = db.Column(db.Integer)
+    restock_level = db.Column(db.Integer)
+    restock_amt = db.Column(db.Integer)
+
+    def __repr__(self):
+        return repr(self.serialize())
+
+    def save(self):
+        """
+        Saves an Entry to database,
+        currently no duplicate detection is supported.
+        """
+        db.session.add(self)
+        db.session.commit()
+
+    def delete(self):
+        """
+        Delete an Entry from database.
+        """
+        db.session.delete(self)
+        db.session.commit()
+
+    def serialize(self):
+        """
+        Serialize an Entry into a dictionary.
+        """
+        return {
+            "prod_id": self.prod_id,
+            "prod_name": self.prod_name,
+            "new_qty": self.new_qty,
+            "used_qty": self.used_qty,
+            "open_boxed_qty": self.open_boxed_qty,
+            "restock_level": self.restock_level,
+            "restock_amt": self.restock_amt
+        }
+
+    def deserialize(self, data, initialize_property=True):
+        """
+        Deserializes an Entry from a dictionary.
+
+        Args:
+            data (dict): A dictionary containing the Entry data
+        """
+        if not isinstance(data, dict):
+            raise DataValidationError('Invalid Entry: body of request contained bad or no data')
+        # only prod_id is a must-have property
+        try:
+            self.prod_id = data['prod_id']
+        except KeyError as error:
+            raise DataValidationError('Invalid Entry: missing ' + error.args[0])
+        except TypeError as error:
+            raise DataValidationError('Invalid Entry: body of request contained' \
+                                      'bad or no data')
+
+        # populate Entry with data or None
+        self.prod_name = data.get('prod_name')
+        self.new_qty = data.get('new_qty')
+        self.used_qty = data.get('used_qty')
+        self.open_boxed_qty = data.get('open_boxed_qty')
+        self.restock_level = data.get('restock_level')
+        self.restock_amt = data.get('restock_amt')
+
+        if initialize_property:
+            if self.prod_name is None:
+                self.prod_name = 'default_prod_name'
+            if self.new_qty is None:
+                self.new_qty = 5
+            if self.used_qty is None:
+                self.used_qty = 0
+            if self.open_boxed_qty is None:
+                self.open_boxed_qty = 0
+            if self.restock_level is None:
+                self.restock_level = 5
+            if self.restock_amt is None:
+                self.restock_amt = 20
+
+        return self
+
+    @staticmethod
+    def init_db(app):
+        """ Initialize database """
+        Entry.app = app
+        db.init_app(app)
+        app.app_context().push()
+        db.create_all()
+
+    @staticmethod
+    def list_all():
```

Comment:
Good job for also adding `list_all()` and `find()`.
But these functions are not under the issue for setting up database, maybe in the future only put code relevant to current issue.

Reply:
Agreed. As I suggested earlier, each PR should focus on one issue. This issue is actually a technical debt, not an "enhancement". Mixing logical code with the framework backbone makes PR review difficult and harder to pass.

Reply:
Advice taken.
Next time I'll try to avoid this situation.